src/main/java/com/faforever/client/fa/ProcessUtils.java

```diff
@@ -0,0 +1,45 @@
+package com.faforever.client.fa;
+
+import com.faforever.client.fa.Kernel32Ex.WindowsPriority;
+import com.sun.jna.Pointer;
+import com.sun.jna.platform.win32.WinDef.BOOL;
+import com.sun.jna.platform.win32.WinDef.DWORD;
+import com.sun.jna.platform.win32.WinNT;
+import com.sun.jna.platform.win32.WinNT.HANDLE;
+import lombok.SneakyThrows;
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+import org.bridj.Platform;
+
+import java.lang.reflect.Field;
+
+@UtilityClass
+@Slf4j
+public class ProcessUtils {
+
+  void setProcessPriority(Process process, WindowsPriority priority) {
+    if (!Platform.isWindows()) {
+      log.debug("Setting process priority is only supported on Windows");
+      return;
+    }
+
+    log.debug("Settings priority of process {} to {}", process.pid(), priority);
+    HANDLE hProcess = getProcessHandle(process);
+    DWORD dwPriorityClass = priority.dword();
+    BOOL success = Kernel32Ex.INSTANCE.SetPriorityClass(hProcess, dwPriorityClass);
+    if (!success.booleanValue()) {
+      log.warn("Could not set priority of process {}", process.pid());
+    }
+  }
+
+  @SneakyThrows
+  private HANDLE getProcessHandle(Process process) {
```

Comment:
Rather than sneaky throws can this catch and either log the exception and cancel setting the priority or catch and rethrow  using some subclass of MajorNotifiableException so that there is a better message given to the user for the error.

Right now if this errors due to some systems permission or something else the user will not have a clear idea what went wrong. And it is likely it may fail on some systems due to security settings. So would be good to either move forward without setting the priority or let the user know with a more informative message.

Reply:
This code can only throw an exception if `ProcessImpl.handle` doesn't exist anymore or if `--add-opens=java.base/java.lang=ALL-UNNAMED` is not set. System permissions would lead to `BOOL success` be false which is already covered. But your point is valid, I refactored it.